testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestHiveSparkCompatibility.java

```diff
@@ -423,6 +424,78 @@ private void testReadTrinoCreatedTable(String tableName, String tableFormat)
         onTrino().executeQuery("DROP TABLE " + trinoTableName);
     }
 
+    @Test(groups = {HIVE_SPARK, PROFILE_SPECIFIC_TESTS})
+    public void testReadSparkInvalidDatePartitionName()
+    {
+        String sparkTableName = "test_trino_reading_spark_date_type_partitioned_" + randomTableSuffix();
+        String trinoTableName = format("%s.default.%s", TRINO_CATALOG, sparkTableName);
+
+        onSpark().executeQuery(format("CREATE TABLE default.%s (value integer) PARTITIONED BY (dt date)", sparkTableName));
+
+        // Spark allows creating partition with invalid date format
+        // Hive deny creating such partitions, but allows reading
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='1965-09-10 23:59:59.999999999') VALUES (1)", sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='1965-09-10 23:59:59') VALUES (2)", sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='1965-09-10 23:59') VALUES (3)", sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='12345-06-07') VALUES (4)", sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='123-04-05') VALUES (5)", sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='-0001-01-01') VALUES (6)", sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='1965-09-10 invalid') VALUES (7)", sparkTableName));
+        onSpark().executeQuery(format("INSERT INTO %s PARTITION(dt='invalid date') VALUES (8)", sparkTableName));
+
+        assertThat(onSpark().executeQuery("SHOW PARTITIONS " + sparkTableName))
+                .containsOnly(List.of(
+                        row("dt=1965-09-10 23%3A59%3A59.999999999"),
+                        row("dt=1965-09-10 23%3A59%3A59"),
+                        row("dt=1965-09-10 23%3A59"),
+                        row("dt=12345-06-07"),
+                        row("dt=123-04-05"),
+                        row("dt=-0001-01-01"),
+                        row("dt=1965-09-10 invalid"),
+                        row("dt=invalid date")));
+
+        // Spark returns incorrect value in case of small digits
+        // Use date_format function to avoid exception due to java.sql.Date.valueOf
+        assertThat(onSpark().executeQuery("SELECT value, dt FROM " + sparkTableName))
+                .containsOnly(List.of(
+                        row(1, "1965-09-10"),
+                        row(2, "1965-09-10"),
+                        row(3, "1965-09-10"),
+                        row(4, "+12345-06-07"),
+                        row(5, null),
+                        row(6, "-0001-01-01"),
+                        row(7, "1965-09-10"),
+                        row(8, null)));
+
+        // Hive returns incorrect value for small dates
+        // Use date_format function to avoid exception due to java.sql.Date.valueOf
```

Comment:
what exception?

Reply:
IllegalArgumentException by 5 digit year. I will update the comment.